### ISSUE TITLE
stairs appear only on layer 6.

### ISSUE THREAD

**Issue by mitm001** [s-leger/archipack#80]
Description of the issue: Archipack stairs quit appearing at cursor on any layer except 6. The origin appears at center on layer 6 without any mesh if adding to any other layer. Only way to work around it is to add layer 6 to the target layer to add the stair at correct position then deselect layer 6 and move the stair to the intended layer. 

Blender version: 2.79c

Os version: win 7 spk 1.

**Comment by s-leger** [comment]
Hi, 
Archipack create objects on active layer (as blender does with other primitives).
So you probalby made layer 6 active before creation.
This will change in 1.4 release, i'm planing a basic layer management - compatible with named layers.

**Comment by mitm001** [comment]
No, Archipack script breaks and will only create the object on one layer, period. In my case it will only add stairs on layer 6. Using any other layer will result in adding the stairs origin point only to layer 6. The mesh is not there or anywhere to be found. Just an origin at center, regardless of where the cursor is or what layer you add it to. 

Not a new user to blender. Just informing you that there is a bug.

**Comment by s-leger** [comment]
Any error reported on console ? 
Could you share failing file, so i'm able to narrow down this issue ?

**Comment by mitm001** [comment]
Wich version of archipack ? 
1.2.8

 Any error reported on console ? 
No. Just the error about missing materials.

 Could you share failing file, so i'm able to narrow down this issue ?
Unfortunately it happened on a file that is proprietary and I cant distribute it.

As a side note, archipack dosen't change object's layers on version < 1.4 so there is something else setting object layer. (1.4 set default layer by object type)

Prior to this happening I was working on a combined layer, ie I had two layers (4, 5) open at the same time by using the shift key to add the second layer. I tested that situation in a new file and had no problems so I don't think it is related.

I had to create a new file and append my objects from the old file to clear it up.
